### PR TITLE
kicad: include desktop, icon and mime files

### LINKED ISSUE #106295: Missing menu entry to launch kicad
**Describe the bug**
`kicad` is missing a menu entry to launch the application.

**To Reproduce**
1. Install `kicad`

**Expected behavior**
There should be a menu entry to launch the application.

**Notify maintainers**

@evils @bgamari

**Metadata**
```
- system: `"x86_64-linux"`
 - host os: `Linux 5.9.12, NixOS, 21.03pre257339.83cbad92d73 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.9`
 - channels(root): `"nixos-21.03pre257339.83cbad92d73"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by eduardosm**
###### Motivation for this change

Fixes https://github.com/NixOS/nixpkgs/issues/106295.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/electronics/kicad/default.nix [hunk 1]
```diff
@@ -227,10 +229,19 @@ stdenv.mkDerivation rec {
...
         # link in the CLI utils
         (map (util: "ln -s ${base}/bin/${util} $out/bin/${util}") utils)
+
+        "runHook postInstall"
       ])
     )
   ;
 
+  postInstall = ''
+    install -d $out/share
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/kicad/default.nix:239]:
```suggestion
    mkdir -p $out/share
```

Comment (eduardosm) [reply pkgs/applications/science/electronics/kicad/default.nix]: Changed

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119055` at 427edc8b run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>kicad-small</li>
    <li>kicad-unstable-small</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`agpl3` is a deprecated license, please check if project uses `agpl3Plus` or `agpl3Only` and change `meta.license` accordingly.

Near pkgs/applications/science/electronics/kicad/default.nix:262:5:
```
    |
262 |     license = lib.licenses.agpl3;
    |     ^
```
</li>
  </ul>
</details>

File: pkgs/applications/science/electronics/kicad/default.nix [hunk 2]
```diff
@@ -216,6 +216,8 @@ stdenv.mkDerivation rec {
     in
     (concatStringsSep "\n"
       (flatten [
+        "runHook preInstall"
```
Comment (evils) [inline pkgs/applications/science/electronics/kicad/default.nix:219]: what's up with this?

File: pkgs/applications/science/electronics/kicad/default.nix [hunk 3]
```diff
@@ -227,10 +229,19 @@ stdenv.mkDerivation rec {
 
         # link in the CLI utils
         (map (util: "ln -s ${base}/bin/${util} $out/bin/${util}") utils)
+
+        "runHook postInstall"
```
Comment (evils) [inline pkgs/applications/science/electronics/kicad/default.nix:233]: any idea why this has to be added explicitly?

**Review by evils — COMMENTED** [review]
sorry i didn't get around to addressing your issue

File: pkgs/applications/science/electronics/kicad/default.nix (hunk 2 above)
Comment (eduardosm) [reply pkgs/applications/science/electronics/kicad/default.nix:219]: Hooks need to be explicitly called by the `installPhase` script. Even if there isn't a `preInstall` hook, users may define their own using `overrideAttrs`.
https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md

**Review by evils — APPROVED** [review]

File: pkgs/applications/science/electronics/kicad/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/applications/science/electronics/kicad/default.nix:233]: Because you are overwriting the phase that normally runs those.

**Review by SuperSandro2000 — APPROVED** [review]
LGTM

~~@evils LGTY?~~ nvm, should have read the above more closely.

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Pull request merged**